#!/usr/bin/env python
PACKAGE = "vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Params for bucket
gen.add("lowH", int_t, 0, "Lower Hue", 52, 0, 255)
gen.add("lowS", int_t, 0, "Lower Sat", 0, 0, 255)
gen.add("lowV", int_t, 0, "Lower Value", 35, 0, 255)
gen.add("hiH", int_t, 0, "High Hue", 143, 0, 255)
gen.add("hiS", int_t, 0, "High Sat", 255, 0, 255)
gen.add("hiV", int_t, 0, "High Value", 255, 0, 255)
gen.add("area_thresh", int_t, 0, "Min area for bucket", 800, 0, 30000)
gen.add("deltaX_multiplier", double_t, 0, "Delta X Multiplier", 3.0, 0, 50.0)
gen.add("sidemove_movement_offset", double_t, 0, "Sidemove Movement Offset", 0.1, 0, 50.0)
gen.add("forward_offset", double_t, 0, "Forward offset", 0.3, 0, 50.0)
gen.add("head_on_area", int_t, 0, "Head on Area", 5000, 0, 30000)

exit(gen.generate(PACKAGE, "flare_vision", "flare"))